File: Fibonacci/fibonacci.py
#!/usr/bin/env python
import sys

def fibonacci(x):
	if x <= 2:
		return x
	return fibonacci(x-1) + fibonacci(x-2)

def fibonacci_memoized(n):
	cache = {0: 1, 1: 1}
	if n in cache:
		return cache[n]
	cache[n] = fibonacci_memoized(n-1) + fibonacci_memoized(n-2)
	return cache[n]

x = int(sys.argv[1])
print fibonacci_memoized(x)

File: Fibonacci/fibonacci.swift
#!/usr/bin/env xcrun swift
func fibonnaci(n: Int) -> Int {
	guard n > 1 else { return n }
	return fibonnaci(n: n-2) + fibonnaci(n: n-1)
}

func fibonnaciMemoized(n: Int) -> Int {
    var cache = [0: 0, 1: 1]
    if let number = cache[n] { 
    	return number
    }
    cache[n] = fibonnaciMemoized(n: n-2) + fibonnaciMemoized(n: n-1)
    return cache[n]!
}

if let argument = Int(CommandLine.arguments[1]) {
	print(fibonnaci(n: argument))
	print(fibonnaciMemoized(n: argument))
}